src/printing.ts

```diff
@@ -0,0 +1,264 @@
+import { NativeModules } from 'react-native';
+
+import lineWrap from 'word-wrap';
+import {
+  PRINTING_ALIGNMENT,
+  PRINTING_COMMANDS,
+  EPOS_BOOLEANS,
+} from './constants';
+
+const { EscPosPrinter } = NativeModules;
+
+type TCommandValue = [key: string, params: any[]];
+type TScalingFactors = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8;
+
+/**
+ * Create an array of commands to send to the printer
+ */
+class Printing {
+  private _buffer: any[];
+  private _state: {
+    bold: boolean;
+    underline: boolean;
+  };
+
+  /**
+   * Create a new object
+   *
+   */
+  constructor() {
+    this._buffer = [];
+    this._state = {
+      bold: false,
+      underline: false,
+    };
+  }
+
+  /**
+   * Reset the state of the object
+   *
+   */
+  _reset() {
+    this._buffer = [];
+
+    this._state = {
+      bold: false,
+      underline: false,
+    };
+  }
+
+  /**
+   * Send the current array of commands to the printer
+   *
+   * @param  {string}   value  String to encode
+   * @return {object}          Encoded string as a ArrayBuffer
+   *
+   */
+  _send(value: any) {
+    return EscPosPrinter.printBuffer(value);
```

Comment:
Could you please add Promise wrapper like for main methods (see getPaperWidth at index.ts ) to provide correct completion for printing task?